### ISSUE TITLE
Settings UI lacks concept of inherited default value

### ISSUE THREAD

**Issue by jrieken** [Microsoft/vscode#57330]
re #57304

* open settings UI
* locate 'Suggest Font Size'
* notice how the UI shows `0` which isn't true - the default value is the font size of the editor. 

<img width="343" alt="screen shot 2018-08-28 at 09 47 01" src="https://user-images.githubusercontent.com/1794099/44708754-32e69500-aaa8-11e8-9088-194a04f4d6bc.png">

**Comment by roblourens** [comment]
0 means "the font size of the editor" but the value itself is 0. If we want to change that, the default could be `null` which would show in the UI as an empty input box. How would you display the real default?

![image](https://user-images.githubusercontent.com/323878/44741050-d9cf2f00-aab0-11e8-954c-448876d6f31b.png)

**Comment by jrieken** [comment]
Well, it should show 12 or whatever the editor font size is. The fact that we internally treat 0 as 'use editor font size' is an implementation that leaks into the UI. I think a user that didn't implement the suggest widget won't understand this

**Comment by roblourens** [comment]
Do you know of any other settings with similar behavior that fall back on the value of another setting?

**Comment by jrieken** [comment]
Yeah, metadata is needed. When defining a setting we should be able to refer to another setting which defines its default

**Comment by roblourens** [comment]
I added a note to the descriptions for those two settings.
